charts/hivemq-platform-operator/templates/deployment.yml

```diff
@@ -51,27 +51,31 @@ spec:
       containers:
         - env:
             - name: JAVA_OPTS
-              value: {{.Values.javaOpts | quote }}
-            {{- if and (ne .Values.selector "nil") (.Values.selector) }}
+              value: {{ .Values.javaOpts | quote }}
+            {{- with .Values.selector }}
             - name: quarkus.operator-sdk.controllers.hivemq-controller.selector
-              value: operator={{ .Values.selector }}
+              value: operator={{ . | trim }}
             {{- end }}
             - name: KUBERNETES_NAMESPACE
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
             - name: hivemq.platform.operator.release.name
               value: "{{ .Release.Name }}"
+            - name: hivemq.platform.operator.crd.apply
+              value: "{{ .Values.crd.apply }}"
```

Comment:
Is the new section `crd` merged correctly with an old custom `values.yaml` that doesn't contain the new default values? Or do we need to handle missing values here?

Reply:
Yes, see previous comment. 
The new environment variables are added and merged into the list. The only drawback of making changes to the environment variable list is that this will trigger a restart of the Platform Operator, automatically managed by the K8s Deployment controller.

Reply:
The only drawback of making changes to the environment variable list is that this will trigger a restart of the Platform Operator, automatically managed by the K8s Deployment controller.

Reply:
I was just unsure about the values merging in nested structures, but it works as it should. You can just configure e.g. `crd.waitReady` and the other `crd.apply` and `crd.waitTimeout` values will be taken from the defaults.

And the operator pod will anyway restart because of the new image/version.

So all is good :)